tests/cpp_tests/test_serialize.cpp

```diff
@@ -0,0 +1,74 @@
+/*!
+ * Copyright (c) 2022 Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE file in the project root for license information.
+ */
+
+#include <gtest/gtest.h>
+#include <testutils.h>
+#include <LightGBM/utils/byte_buffer.h>
+#include <LightGBM/utils/log.h>
+#include <LightGBM/c_api.h>
+#include <LightGBM/dataset.h>
+
+#include <iostream>
+
+using LightGBM::ByteBuffer;
+using LightGBM::Dataset;
+using LightGBM::Log;
+using LightGBM::TestUtils;
+
+TEST(Serialization, JustWorks) {
+  // Load some test data
+  DatasetHandle datset_handle;
+  const char* params = "max_bin=15";
+  int result = TestUtils::LoadDatasetFromExamples("binary_classification/binary.train", params, &datset_handle);
+  EXPECT_EQ(0, result) << "LoadDatasetFromExamples result code: " << result;
+
+  Dataset* dataset;
+  try {
```

Comment:
Doesn't `try/catch` here mean that test can silently fail and continue producing green CI job indicator even when everything is actually broken?

Reply:
ah, good catch, I had added that last minute to handle memory freeing.

Reply:
OK, thanks for updating tests!
I guess we need the same in streaming tests:
https://github.com/microsoft/LightGBM/blob/c8712a9e083bfa0922f6e860f49a8a374816f43e/tests/cpp_tests/test_stream.cpp#L38
https://github.com/microsoft/LightGBM/blob/c8712a9e083bfa0922f6e860f49a8a374816f43e/tests/cpp_tests/test_stream.cpp#L145